core/network/impl/synchronizer_impl.cpp

```diff
@@ -59,16 +61,69 @@
     app_state_manager->atShutdown([this] { node_is_shutting_down_ = true; });
   }
 
+  bool SynchronizerImpl::subscribeToBlock(
+      const primitives::BlockInfo &block_info, SyncResultHandler &&handler) {
+    // Check if block is already in tree
+    if (block_tree_->hasBlockHeader(block_info.hash)) {
+      handler(block_info);
```

Comment:
hmm, shouldn't we send here something like Error::BLOCK_ALREADY_ARRIVED ?

Reply:
I want to refactor it to use new-added mechanism of subscription instead propagating of handler through function calls.
In this particular case must be positive result for correct working.